extensions/amp-story/1.0/bookend/amp-story-bookend.js

```diff
@@ -379,10 +302,10 @@ export class Bookend {
    * will prerender the bookend DOM, but there are cases where we need it before
    * the component is built. Eg: the desktop share button needs the providers.
    * @param {boolean=} applyConfig  Whether the config should be set.
-   * @return {!Promise<(?BookendConfigDef|?./bookend-component.BookendDataDef)>}
+   * @return {!Promise<?./bookend-component.BookendDataDef>}
    */
   loadConfig(applyConfig = true) {
-    if (this.config_ !== undefined) {
+    if (this.config_) {
```

Comment:
This is getting a little messy. I bet you could simplify a bit (and make Closure complain less) if you split `loadConfig` and `applyConfig` as two separate methods.

`loadConfig` returns a promise (which might already be resolved). `applyConfig` calls `loadConfig`. Simpler and keeps you from checking config state.

Reply:
sounds good!